File: WebApplication/App/BusinessLogic/Entities/Note.ts
import * as _ie from "./IEntity";

export namespace TigerSpike.Test.BusinessLogic.Entities {
    export class Note implements _ie.TigerSpike.Test.BusinessLogic.Entities.IEntity<Note> {
        public Username: string;
        public Location: Position;
        public Text: string;

        public Construct(data: any): Note {
            this.Username = data.Username;
            this.Location = data.Location;
            this.Text = data.Text;
            return this;
        }
    }
}

File: BusinessLogic/Entities/Note.cs
using System.Data;
using BusinessLogic.Helpers.Extensions.DataRow;

namespace BusinessLogic.Entities
{
    public class Note : IEntity<Note>
    {
        #region Attributes
        private string _username;
        private string _text;
        private Position _location;
        #endregion

        #region Properties

        public Position Location { get => _location; set => _location = value; }
        public string Text { get => _text; set => _text = value; }
        public string Username { get => _username; set => _username = value; }
        #endregion

        #region Methods
        public Note()
        {
            _username = string.Empty;
            _text = string.Empty;
            _location = new Position();
            _location.coords = new Position.Coordinates();
        }

        public Note Construct(DataRow row)
        {
            _username = row.GetValue<string>("username", string.Empty);
            _text = row.GetValue<string>("text", string.Empty);
            _location.coords.latitude = row.GetValue<decimal>("lat");
            _location.coords.longitude = row.GetValue<decimal>("lon");

            return this;
        }
        #endregion
    }
}
